### PR TITLE
cmd/clef: add newaccount command

### PR DESCRIPTION
**Pull request by holiman**
This makes it possible to create accounts via CLI. Since we don't have any good UIs, what the heck

### CODE REVIEW

**Comment by holiman** [comment]
Usage:
```
[user@work go-ethereum]$ build/bin/clef  newaccount

WARNING!

Clef is an account management tool. It may, like any software, contain bugs.

Please take care to
- backup your keystore files,
- verify that the keystore(s) can be opened with your password.

Clef is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
PURPOSE. See the GNU General Public License for more details.

Enter 'ok' to proceed:
> ok

INFO [03-18|22:01:16.781] Starting clef                            keystore=/home/user/.ethereum/keystore light-kdf=false
DEBUG[03-18|22:01:16.788] FS scan times                            list=6.796992ms set=56.822µs diff=9.374µs
-------- New Account request--------------

A request has been made to create a new account. 
Approving this operation means that a new account is created,
and the address is returned to the external caller

Request context:
        NA -> NA -> NA

Additional HTTP header data, provided by the external caller:
        User-Agent: 
        Origin: 
Approve? [y/N]:
> y
## New account password

Please enter a password for the new account to be created (attempt 0 of 3)
> -----------------------
DEBUG[03-18|22:01:22.650] FS scan times                            list=120.578µs  set=46.392µs diff=6.58µs
INFO [03-18|22:01:22.945] Your new key was generated               address=0xFd0fB88d8976eB6d766F742a47FA7fC850531269
WARN [03-18|22:01:22.945] Please backup your key file!             path=/home/user/.ethereum/keystore/UTC--2020-03-18T21-01-21.377431277Z--fd0fb88d8976eb6d766f742a47fa7fc850531269
WARN [03-18|22:01:22.945] Please remember your password! 
Generated account 0xFd0fB88d8976eB6d766F742a47FA7fC850531269
```

**Comment by holiman** [comment]
Actually, this whole dance with "approve?" is silly, from a UX perspective. I'll fix it.

**Comment by holiman** [comment]
Failure case: 
```
[user@work go-ethereum]$ build/bin/clef newaccount

WARNING!

Clef is an account management tool. It may, like any software, contain bugs.

Please take care to
- backup your keystore files,
- verify that the keystore(s) can be opened with your password.

Clef is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
PURPOSE. See the GNU General Public License for more details.

Enter 'ok' to proceed:
> ok

INFO [03-19|10:22:14.540] Starting clef                            keystore=/home/user/.ethereum/keystore light-kdf=false
DEBUG[03-19|10:22:14.547] FS scan times                            list=6.784652ms set=87.912µs diff=9.257µs
## New account password

Please enter a password for the new account to be created (attempt 0 of 3)
> -----------------------
## Error 
Account creation attempt #1 failed due to password requirements: password too short (<10 characters)
-------------------------------------------
## New account password

Please enter a password for the new account to be created (attempt 1 of 3)
> -----------------------
## Error 
Account creation attempt #2 failed due to password requirements: password too short (<10 characters)
-------------------------------------------
## New account password

Please enter a password for the new account to be created (attempt 2 of 3)
> -----------------------
## Error 
Account creation attempt #3 failed due to password requirements: password too short (<10 characters)
-------------------------------------------
account creation failed
```
Success case: 
```
[user@work go-ethereum]$ build/bin/clef newaccount

WARNING!

Clef is an account management tool. It may, like any software, contain bugs.

Please take care to
- backup your keystore files,
- verify that the keystore(s) can be opened with your password.

Clef is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
PURPOSE. See the GNU General Public License for more details.

Enter 'ok' to proceed:
> ok  

INFO [03-19|10:22:23.413] Starting clef                            keystore=/home/user/.ethereum/keystore light-kdf=false
DEBUG[03-19|10:22:23.413] FS scan times                            list=109.535µs set=49.683µs diff=11.945µs
## New account password

Please enter a password for the new account to be created (attempt 0 of 3)
> -----------------------
DEBUG[03-19|10:22:27.391] FS scan times                            list=122.682µs set=49.366µs diff=7.761µs
INFO [03-19|10:22:27.702] Your new key was generated               address=0xb9a900968016E3B693EFA4129963EfadE1A840B9
WARN [03-19|10:22:27.702] Please backup your key file!             path=/home/user/.ethereum/keystore/UTC--2020-03-19T09-22-26.192676878Z--b9a900968016e3b693efa4129963efade1a840b9
WARN [03-19|10:22:27.702] Please remember your password! 
Generated account 0xb9a900968016E3B693EFA4129963EfadE1A840B9
[user@work go-ethereum]$ 

```

**Comment by holiman** [comment]
@fjl could you take a look at this please?

File: cmd/clef/intapi_changelog.md [hunk 1]
```diff
@@ -10,6 +10,17 @@ TL;DR: Given a version number MAJOR.MINOR.PATCH, increment the:
 
 Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
 
+### 7.0.1 
+
+Added `clef_New` to the internal API calleable from a UI. 
+
+> `New` creates a new password protected Account. The private key is protected with
+> the given password. Users are responsible to backup the private key that is stored
+> in the keystore location thas was specified when this API was created.
```
Comment (ligi) [inline cmd/clef/intapi_changelog.md]:
```suggestion
> in the keystore location that was specified when this API was created.
```

File: signer/core/uiapi.go [hunk 2]
```diff
@@ -195,6 +195,16 @@ func (api *UIServerAPI) Import(ctx context.Context, keyJSON json.RawMessage, old
 	return be[0].(*keystore.KeyStore).Import(keyJSON, oldPassphrase, newPassphrase)
 }
 
+// New creates a new password protected Account. The private key is protected with
+// the given password. Users are responsible to backup the private key that is stored
+// in the keystore location thas was specified when this API was created.
```
Comment (ligi) [inline signer/core/uiapi.go]:
```suggestion
// in the keystore location that was specified when this API was created.
```

**Pull request merged**